### ISSUE TITLE
[BUG] Crash when generation is terminated immediately

### ISSUE THREAD

**Issue by Nabushika** [theroyallab/tabbyAPI#91]
**Describe the bug**
Crashes when generation produces only a finish reason and nothing else

**To Reproduce**
1. Load up TabbyAPI with a model
2. Submit a generation request that results in the model producing stop token immediately

**Expected behavior**
Empty generation returned

**Logs**
`generations = [{'finish_reason': 'stop'}]` at `backends/exllamav2/model.py:596`
Request is 
```
{
            "model": "...",
            "messages":  "Hello, how can I help you today?",
            "temperature": 0.8,
            "max_tokens": 512,
            "stop": "\n",
}
```
Note the stop token is probably expected after the prompt I submitted - after changing the stop token, completion worked fine.
```
ERROR:    Traceback (most recent call last):
ERROR:      File "/home/abel/tabbyAPI/endpoints/OAI/utils/chat_completion.py", line 193, in 
generate_chat_completion
ERROR:        generation = await model.container.generate(
ERROR:      File "/home/abel/tabbyAPI/backends/exllamav2/model.py", line 619, in generate
ERROR:        generations[-1].get("prompt_tokens"), 0
ERROR:    IndexError: list index out of range
ERROR:    Sent to request: Chat completion aborted. Maybe the model was unloaded? Please check the 
server console.
INFO:     127.0.0.1:54892 - "POST /v1/chat/completions HTTP/1.1" 503
```

**System info** (Bugs without this information will go lower on our priority list!)
 - OS: Linux
 - Python version: 3.10
 - CUDA/ROCm version: 12.1

**Additional context**
Caused by line 518, `finish_reason_gen = generations.pop()` then later on 618, `unwrap(generations[-1].get("prompt_tokens"), 0)`
Yes I know this is an edge case, I'm going to work around it for now, but I'd been working on something that interacts with the API and I thought it was my request that was still bad.
Should be an easy fix, splitting the `if` condition on line 595 into two separate if conditions that check the same thing, might submit a PR but am focused on something else for now.

**Comment by bdashore3** [comment]
Should be fixed in 190a0b26c36c0e932ceda7ca5e25b123f22d695a

**Comment by bdashore3** [comment]
This bug has been fixed and tested, so I'm closing this issue.
